pints/_plot.py

```diff
@@ -0,0 +1,248 @@
+#
+# Quick diagnostic tools for inference results
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import pints
+import numpy as np
+from scipy import stats
+#import matplotlib.pyplot as plt
+import pylab as plt
+
+def trace(chain):
+    """
+    Trace plot.
+
+    `chain`
+        MCMC routine returned chain
+
+    """
+    n_sample, n_param = chain.shape
+
+    fig = plt.figure(figsize=(12, 2*n_param))
+    for i in xrange(n_param):
+        # Add histogram subplot
+        plt.subplot(n_param, 2, 1 + 2 * i)
+        plt.xlabel('Parameter ' + str(i + 1))
+        plt.ylabel('Frequency')
+        plt.hist(chain[:,i], label='p' + str(i + 1), bins=40, color='green',
+        alpha=0.5)
+
+        # Add trace subplot
+        plt.subplot(n_param, 2, 2 + 2 * i)
+        plt.xlabel('Iteration')
+        plt.ylabel('Parameter ' + str(i + 1))
+        plt.plot(chain[:,i], color='green', alpha=0.5)
+
+    plt.tight_layout()
+    return fig
+
+def autocorrelation(chain):
+    """
+    Autocorrelation plot for MCMC.
+
+    `chain`
+        MCMC routine returned chain
+
+    """
+    n_sample, n_param = chain.shape
+
+    fig, axes = plt.subplots(n_param, 1, sharex=True, figsize=(6, 2*n_param))
+    for i in xrange(n_param):
+        axes[i].acorr(chain[:,i] - np.mean(chain[:,i]), label='p'+str(i+1))
+        axes[i].set_xlim(-0.5, 20.5)
+    axes[i].set_xlabel('Lag')
+    fig.text(0.04, 0.5, 'Autocorrelation', va='center', rotation='vertical')
+
+    return fig
+
+def prediction(chain, problem):
+    """
+    Predicted time series plot.
+
+    `chain`
+        MCMC routine returned chain
+    `problem`
+        pints.SingleSeriesProblem object
+
+    """
+    if not isinstance(problem, pints.SingleSeriesProblem):
+        raise TypeError('pints.SingleSeriesProblem type is expected')
```

Comment:
Please make a little more helpful (it's 99% of the way there). "First argument to prediction() must be " or something. Error messages are the main place a library interacts with its users...